packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -372,6 +372,46 @@ export default class ChildProcessWorker
         this._child.send(this._request);
       }
     } else {
+      // At this point, it's not clear why the child process exited. There could
+      // be several reasons:
```

Comment:
While we can reasonably guess when the Node.js internal max heap limit has been hit, it's difficult to guess other failure scenarios from `exitCode` and `signal`.

There was a previous discussion of this in the linked issue: https://github.com/facebook/jest/issues/13183#issuecomment-1229149544

I feel propagating an error in the `else` block is the best way to handle unknown situations.